Installing zim-tools fails on Ubuntu Server 20.04.2 -- partial workaround is to download mustache.hpp into /usr/include

Summary: zim-tools install instructions at https://github.com/openzim/zim-tools#compilation do not work on Ubuntu Server 20.04.2

_Thanks if anybody has suggestions for PROBLEM 2 OF 2 below!_

### PROBLEM 1 OF 2:

```
git clone https://github.com/openzim/zim-tools
cd zim-tools
meson . build
```

...can be solved with this workaround: (thanks to @curtathompson and @georgejhunt)

```
cd /usr/include
wget https://raw.githubusercontent.com/kainjow/Mustache/master/mustache.hpp
cd /opt/iiab/zim-tools
meson . build
```

Error Log 1: (output of `meson . build`)
https://paste.ubuntu.com/p/wZkcfSHsTm/

Error Log 2: (/opt/iiab/zim-tools/build/meson-logs/meson-log.txt)
https://paste.ubuntu.com/p/HK45rXKqFS/

### PROBLEM 2 OF 2: (this is the output of `ninja -C build`)

```
ninja: Entering directory `build'
[2/22] Compiling C++ object 'src/25a6634@@zimsearch@exe/zimsearch.cpp.o'.
FAILED: src/25a6634@@zimsearch@exe/zimsearch.cpp.o
c++ -Isrc/25a6634@@zimsearch@exe -Isrc -I../src -I/usr/local/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Werror -std=c++11 -g -Werror -Wall '-DVERSION="2.2.0"' -MD -MQ 'src/25a6634@@zimsearch@exe/zimsearch.cpp.o' -MF 'src/25a6634@@zimsearch@exe/zimsearch.cpp.o.d' -o 'src/25a6634@@zimsearch@exe/zimsearch.cpp.o' -c ../src/zimsearch.cpp
../src/zimsearch.cpp: In function ‘void printSearchResults(zim::Search&)’:
../src/zimsearch.cpp:28:23: error: ‘iterator’ is not a member of ‘zim::Search’
   28 |     for (zim::Search::iterator it = search.begin(); it != search.end(); ++it)
      |                       ^~~~~~~~
../src/zimsearch.cpp:28:53: error: ‘it’ was not declared in this scope; did you mean ‘int’?
   28 |     for (zim::Search::iterator it = search.begin(); it != search.end(); ++it)
      |                                                     ^~
      |                                                     int
../src/zimsearch.cpp:28:66: error: ‘class zim::Search’ has no member named ‘end’
   28 |     for (zim::Search::iterator it = search.begin(); it != search.end(); ++it)
      |                                                                  ^~~
../src/zimsearch.cpp: In function ‘int main(int, char**)’:
../src/zimsearch.cpp:65:34: error: no matching function for call to ‘zim::Search::Search(zim::Archive&)’
   65 |     zim::Search search(zimarchive);
      |                                  ^
In file included from ../src/zimsearch.cpp:21:
/usr/local/include/zim/search.h:182:9: note: candidate: ‘zim::Search::Search(std::shared_ptr<zim::InternalDataBase>, const zim::Query&)’
  182 |         Search(std::shared_ptr<InternalDataBase> p_internalDb, const Query& query);
      |         ^~~~~~
/usr/local/include/zim/search.h:182:9: note:   candidate expects 2 arguments, 1 provided
/usr/local/include/zim/search.h:162:9: note: candidate: ‘zim::Search::Search(zim::Search&&)’
  162 |         Search(Search&& s);
      |         ^~~~~~
/usr/local/include/zim/search.h:162:25: note:   no known conversion for argument 1 from ‘zim::Archive’ to ‘zim::Search&&’
  162 |         Search(Search&& s);
      |                ~~~~~~~~~^
../src/zimsearch.cpp:66:12: error: ‘class zim::Search’ has no member named ‘set_query’; did you mean ‘zim::Query zim::Search::m_query’? (not accessible from this context)
   66 |     search.set_query(s);
      |            ^~~~~~~~~
In file included from ../src/zimsearch.cpp:21:
/usr/local/include/zim/search.h:188:16: note: declared private here
  188 |          Query m_query;
      |                ^~~~~~~
[3/22] Compiling C++ object 'src/25a6634@@zimpatch@exe/zimpatch.cpp.o'.
FAILED: src/25a6634@@zimpatch@exe/zimpatch.cpp.o
c++ -Isrc/25a6634@@zimpatch@exe -Isrc -I../src -I/usr/local/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Werror -std=c++11 -g -Werror -Wall '-DVERSION="2.2.0"' -MD -MQ 'src/25a6634@@zimpatch@exe/zimpatch.cpp.o' -MF 'src/25a6634@@zimpatch@exe/zimpatch.cpp.o.d' -o 'src/25a6634@@zimpatch@exe/zimpatch.cpp.o' -c ../src/zimpatch.cpp
../src/zimpatch.cpp: In function ‘void create(const string&, const string&, const string&)’:
../src/zimpatch.cpp:67:14: error: ‘class zim::writer::Creator’ has no member named ‘configMinClusterSize’; did you mean ‘configClusterSize’?
   67 |   zimCreator.configMinClusterSize(2048);
      |              ^~~~~~~~~~~~~~~~~~~~
      |              configClusterSize
[4/22] Compiling C++ object 'src/25a6634@@zimpatch@exe/tools.cpp.o'.
ninja: build stopped: subcommand failed.
```

--- a/README.md
+++ b/README.md
@@ -42,12 +42,11 @@ with the ZIM libary compilation itself, we recommend to have a look to
 Dependencies
 ------------
 
-Most of the utilities that form the zim-tools programs rely on libzim:
-
 * [ZIM](https://openzim.org) (package `libzim-dev` on Debian/Ubuntu)
-
-`zimdump` additionally requires:
 * [docopt.cpp](https://github.com/docopt/docopt.cpp) (package `libdocopt-dev` on Debian/Ubuntu)
+* [Mustache](https://github.com/kainjow/Mustache) (package `libkainjow-mustache-dev` on Debian/Ubuntu)
+  Be sure you use Mustache version 4.1 or above. You can just copy the header `mustache.hpp`
+  somewhere it can be found by the compiler and/or set CPPFLAGS with correct `-I` option.
 
 `zimwriterfs` relies on many third-party software libraries. They are
 prerequisites to compiling zimwriterfs. The following libraries

